tgui/packages/tgui/interfaces/MusicInstrument.tsx

```diff
@@ -0,0 +1,183 @@
+import { useBackend, useLocalState } from '../backend';
+import { Button, Flex, LabeledList, Box, Knob, Input } from '../components';
+import { Window } from '../layouts';
+
+type MusicInstrumentData = {
+  name: string;
+  notes: string[];
+};
+
+export const MusicInstrument = (props, context) => {
+  const { act, data } = useBackend<MusicInstrumentData>(context);
+  const { name, notes } = data;
+
+  const [noteKeysOrder, setNoteKeysOrder] = useLocalState(
+    context,
+    'keyboardBindingsOrder',
+    'zsxdcvgbhnjmq2w3er5t6y7ui9o0p+'.split('')
+  );
+
+  const [activeKeys, setActiveKeys] = useLocalState(context, 'keyboardActivekeys', new Array(notes.length));
+  const [keyOffset, setKeyOffset] = useLocalState(context, 'keyOffset', 0);
+  const [keybindToggle, setKeybindToggle] = useLocalState(context, 'keybindToggle', false);
+  const [volume, setVolume] = useLocalState(context, 'keyboardVolume', 50);
+  const [transpose, setTranspose] = useLocalState(context, 'keyboardTranspose', 0);
+
+  const toggleKeybind = () => {
+    if (keybindToggle) {
+      act('play_keyboard_off');
+      setKeybindToggle(false);
```

Comment:
Combo of the `act` and the local setter here is indicative that this should probably really be `ui_data`. I know the `act` bit is to toggle a back-end thing, but I could easily see these two bits getting out of sync if not using a single source of truth.

Reply:
Makes sense, I'll change it

Reply:
@mordent-goonstation how should I properly track a per-client setting in `ui_data`?

Reply:
Made so that there is a single variable in instruments.dm and reflects to tgui, modified by ui_act